ui/app/components/auth-saml.js

```diff
@@ -17,83 +19,79 @@ const ERROR_MISSING_PARAMS =
   'The callback from the provider did not supply all of the required parameters. Please click Sign In to try again. If the problem persists, you may want to contact your administrator.';
 export { ERROR_WINDOW_CLOSED, ERROR_MISSING_PARAMS };
 
-export default Component.extend({
-  store: service(),
-  featureFlagService: service('featureFlag'),
+export default class AuthSaml extends Component {
+  @service store;
+  @service featureFlag;
 
-  selectedAuthPath: null,
-  selectedAuthType: null,
-  roleName: null,
-  errorMessage: null,
-  onRoleName() {},
-  onLoading() {},
-  onError() {},
-  onNamespace() {},
+  @tracked errorMessage;
 
-  didReceiveAttrs() {
```

Comment:
`didReceiveAttrs()` runs on init and re-rendering - so I'm not sure replacing with the `constructor()` will give us the same behavior. It seems like the purpose is to reset the error message on init and when any of the args change? [docs](https://guides.emberjs.com/v3.4.0/components/the-component-lifecycle/#toc_formatting-component-attributes-with-didreceiveattrs)

Reply:
I _think_ this can just be removed, since the tracked variable initializes as undefined above and then it looks like below `this.onError(null)` handles resetting the error to null?

Reply:
Yeah, you're right. It resets it to null when you try again. Thanks for the feedback. i removed the constructor.